addons/product/migrations/13.0.1.2/pre-migration.py

```diff
@@ -134,38 +171,76 @@ def calculate_product_product_combination_indices(env):
+                STRING_AGG(ptav.id::varchar, ',' ORDER BY ptav.id) indices
             FROM product_attribute_value_product_product_rel pavppr
             JOIN product_product pp ON pp.id = pavppr.product_product_id
             JOIN product_template_attribute_value ptav
-                ON ptav.product_attribute_value_id =
+                ON (ptav.product_attribute_value_id =
                     pavppr.product_attribute_value_id
-                AND pp.product_tmpl_id = ptav.product_tmpl_id
-        )
-        UPDATE product_product pp
-        SET combination_indices = grouped_pvc.indices
-        FROM (
-            SELECT pvc.product_product_id, STRING_AGG(
-                pvc.product_template_attribute_value_id::varchar, ','
-                ORDER BY pvc.product_template_attribute_value_id) indices
-            FROM pvc
-            JOIN product_product pp ON pp.id = pvc.product_product_id
-            GROUP BY pvc.product_product_id
+                AND pp.product_tmpl_id = ptav.product_tmpl_id)
+            GROUP BY pavppr.product_product_id
         ) grouped_pvc
-        WHERE grouped_pvc.product_product_id = pp.id
-        """,
+        WHERE grouped_pvc.product_product_id = pp.id""",
     )
 
 
-def add_product_template_attribute_value__attribute_id_column(env):
-    """For avoiding that ORM fills it. We fill it later on post-migration."""
+def fill_product_template_attribute_value_attribute_line_id(env):
+    """Done in pre because the field attribute_line_id of ptav is required.
+    As this field is already filled in archived ptavs (in missing ptavs query),
+    we fill it here for active ptavs"""
+    openupgrade.logged_query(
+        env.cr,
+        "ALTER TABLE product_template_attribute_value "
+        "ADD COLUMN attribute_line_id INT4",
+    )
+    # use ptal.active = TRUE
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE product_template_attribute_value ptav
+        SET attribute_line_id = ptal.id
+        FROM product_template_attribute_line ptal
+        JOIN product_template pt ON ptal.product_tmpl_id = pt.id
+        JOIN product_attribute_value pav ON ptal.attribute_id = pav.attribute_id
+        WHERE ptal.active = TRUE AND ptav.product_tmpl_id = pt.id
+            AND ptav.product_attribute_value_id = pav.id
+            AND ptav.attribute_line_id IS NULL""",
+    )
+    # use ptal.active != TRUE
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE product_template_attribute_value ptav
+        SET attribute_line_id = ptal.id
+        FROM product_template_attribute_line ptal
+        JOIN product_template pt ON ptal.product_tmpl_id = pt.id
+        JOIN product_attribute_value pav ON ptal.attribute_id = pav.attribute_id
+        WHERE ptal.active IS DISTINCT FROM TRUE AND
+            ptav.product_tmpl_id = pt.id AND
+            ptav.product_attribute_value_id = pav.id
+            AND ptav.attribute_line_id IS NULL""",
+    )
```

Comment:
Now we can do it in one query:
<pre>
  openupgrade.logged_query(
      env.cr, """
      UPDATE product_template_attribute_value ptav
      SET attribute_line_id = ptal.id
      FROM product_template_attribute_line ptal
      JOIN product_attribute_value pav ON ptal.attribute_id = pav.attribute_id
      WHERE ptav.product_tmpl_id = ptal.product_tmpl_id
          AND ptav.product_attribute_value_id = pav.id
          AND ptav.attribute_line_id IS NULL""",
  )
</pre>